tests/ops/qubit/test_parametric_ops.py

```diff
@@ -25,6 +25,7 @@
 
 import pennylane as qml
 from pennylane import numpy as npp
+from pennylane.ops.functions.assert_valid import _test_decomposition_rule
```

Comment:
Why is `test_decomposition_rule` private?

Reply:
All individual tests in `assert_valid` are private.

Reply:
We only do so in tests.

Reply:
Yes, but tests aren't part of pennylane source code. We're not testing a private function, which I would consider to be fine; we're using a private function to test something else.